File: app/src/main/java/com/gaugustini/mhfudatabase/data/model/Decoration.kt
package com.gaugustini.mhfudatabase.data.model

import com.gaugustini.mhfudatabase.data.enums.ItemIconColor
import com.gaugustini.mhfudatabase.data.enums.ItemIconType

data class Decoration(
    val id: Int,
    val name: String,
    val description: String,
    val rarity: Int,
    val requiredSlots: Int,
    val buyPrice: Int,
    val sellPrice: Int,
    val iconType: ItemIconType,
    val iconColor: ItemIconColor,
)

data class DecorationDetails(
    val decoration: Decoration,
    val skills: List<SkillTreePoints>,
    val recipeA: List<ItemQuantity>,
    val recipeB: List<ItemQuantity>,
)

File: lib/features/decoration/domain/decoration.dart
import '../../../core/domain/enums.dart';
import '../../../core/domain/shared.dart';

class const Decoration({
  required final int id,
  required final String name,
  required final String description,
  required final int rarity,
  required final int buyPrice,
  required final int sellPrice,
  required final int requiredSlots,
  required final ItemIconColor color,
  final List<SkillPoint>? skills,
  final List<ItemQuantity>? recipeA,
  final List<ItemQuantity>? recipeB,
});
